stock_picking_invoicing/models/stock_move.py

```diff
@@ -0,0 +1,65 @@
+# Copyright (C) 2019-Today: Odoo Community Association (OCA)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class StockMove(models.Model):
+    _name = "stock.move"
+    _inherit = [
+        _name,
+        "stock.invoice.state.mixin",
+    ]
+
+    def _get_taxes(self, fiscal_position, inv_type):
+        """
+        Map product taxes based on given fiscal position
+        :param fiscal_position: account.fiscal.position recordset
+        :param inv_type: string
+        :return: account.tax recordset
+        """
+        product = self.mapped("product_id")
+        product.ensure_one()
+        if inv_type in ("out_invoice", "out_refund"):
+            taxes = product.taxes_id
+        else:
+            taxes = product.supplier_taxes_id
+        company_id = self.env.context.get("force_company", self.env.user.company_id.id)
+        my_taxes = taxes.filtered(lambda r: r.company_id.id == company_id)
+        return fiscal_position.map_tax(my_taxes)
+
+    @api.model
+    def _get_account(self, fiscal_position, account):
+        """
+        Map the given account with given fiscal position
+        :param fiscal_position: account.fiscal.position recordset
+        :param account: account.account recordset
+        :return: account.account recordset
+        """
+        return fiscal_position.map_account(account)
+
+    def _get_price_unit_invoice(self, inv_type, partner, qty=1):
+        """
+        Gets price unit for invoice
+        :param inv_type: str
+        :param partner: res.partner
+        :param qty: float
+        :return: float
+        """
+        product = self.mapped("product_id")
+        product.ensure_one()
+        if inv_type in ("in_invoice", "in_refund"):
+            result = product.price
+        else:
+            # If partner given, search price in its sale pricelist
+            if partner and partner.property_product_pricelist:
+                product = product.with_context(
```

Comment:
This piece of code won't work if the pricelist is configured to show the discount to the customer.

Reply:
I think I understand what you mean. Would using `def get_product_price(self, product, quantity, partner, date=False, uom_id=False)` of `product.pricelist` to get the price solve this?

Reply:
you should have a look here: https://github.com/odoo/odoo/blob/ae2d4b8060fe4c6c6ebf04739d33bb150b923224/addons/sale/models/sale.py#L1750

Reply:
@Cedric-Pigeon @Rad0van the case of use for me is Brazilian Localization, in this case the creation of invoice from stock.picking means the company should created a Fiscal Document/account.invoice for any product outside movement included between the same company but in different address and another cases where are not an Purchase or Sale Order, so in those case the Price should be the Product Standard (  brazilian localization change for this case https://github.com/OCA/l10n-brazil/blob/12.0/l10n_br_stock_account/models/stock_move.py#L131 ). To use the creation of invoice from stock.picking integrated with Sale and Purchase and put back the possibilite to force the creation of invoice only from stock.picking( like the old behaivor in v8 ) was created two modules but at brazilian localization https://github.com/OCA/l10n-brazil/tree/12.0/l10n_br_sale_stock https://github.com/OCA/l10n-brazil/tree/12.0/l10n_br_purchase_stock ( we think in extract parts of the modules from there and put in this repo, but in v14) and in those modules the price_unit used are the value of price_unit at Purchase or Sale line( where the question about discount are already treat ).  It is important hear about the other cases of use and what the value of price_unit should expect here or even have different values depending of the kind of the movement .Do you think we should change the calcule ? Something like this:
```
if partner and partner.property_product_pricelist:
    product = product.with_context(
        lang=partner.lang,
        partner=partner.id,
        quantity=qty,
        pricelist=partner.property_product_pricelist.id,
        uom=fields.first(self).product_uom.id,
        date_order=fields.first(self).picking_id.scheduled_date,
    )
    #
    company = fields.first(self).env.user.company_id
    Fpos_sudo = fields.first(self).env['account.fiscal.position'].sudo()
    fpos_id = Fpos_sudo.with_context(
         force_company=company.id).get_fiscal_position(partner.id)
    taxes = Fpos_sudo.browse(fpos_id).map_tax(
         product.sudo().taxes_id.filtered(
             lambda x: x.company_id == company), product, partner)
    tax_obj = fields.first(self).env['account.tax']
    price_unit = tax_obj._fix_tax_included_price_company(
        product.price, product.taxes_id, taxes,
        fields.first(self).company_id)
    result = price_unit
```
Should we include field for discount, or another one, in stock.move ?